nixos/modules/services/web-apps/comfyui.nix

```diff
@@ -0,0 +1,498 @@
+        default = false;
+        description = "Whether or not to enable ROCM for ATi GPU acceleration.";
+        defaultText = literalExpression "false";
+        example = literalExpression "true";
+      };
+
+      package = mkOption {
+        type = types.package;
+        default = (
+          if config.cudaSupport
+          then pkgs.comfyui-cuda
+          else if config.rocmSupport
+          then pkgs.comfyui-rocm
+          else pkgs.comfyui-cpu
+        );
+        defaultText = literalExpression "pkgs.comfyui";
+        example = literalExpression "pkgs.comfyui-rocm";
+        description = "ComfyUI base package to use.";
+      };
+
+      user = mkOption {
+        type = types.str;
+        default = defaultUser;
+        example = "yourUser";
+        description = ''
+          The user to run ComfyUI as.
+          By default, a user named `${defaultUser}` will be created whose home
+          directory will contain input, output, custom nodes and models.
+        '';
+      };
+
+      group = mkOption {
+        type = types.str;
+        default = defaultGroup;
+        example = "yourGroup";
+        description = ''
+          The group to run ComfyUI as.
+          By default, a group named `${defaultUser}` will be created.
+        '';
+      };
+
+      useCPU = mkOption {
+        type = types.bool;
+        default = false;
+        description = ''
+          Uses the CPU for everything. Very slow, but needed if there is no hardware acceleration.
+        '';
+      };
+
+      port = mkOption {
+        type = types.port;
+        default = 8188;
+        description = "Set the listen port for the Web UI and API.";
+      };
+
+      customNodes = mkOption {
+        type = types.listOf types.package;
+        default = [];
+        description = "custom nodes to add to the ComfyUI setup. Expects a list of packages from pkgs.comfyui-custom-nodes";
+      };
+
+      listen = mkOption {
+        type = types.str;
+        # Assume a higher security posture by default.
+        default = "127.0.0.1";
+        description = "The net mask to listen to.";
+        example = "0.0.0.0";
+      };
+
+      cors-origin-domain = mkOption {
+        type = types.str;
+        default = "disabled";
+        description = ''The CORS domain to bless.  Use "disabled" to disable.  This must include a port'';
+        example = "foo.com:443";
+      };
+
+      cuda-malloc = mkOption {
+        type = types.bool;
+        default = cfg.cudaSupport;
+      };
+
+      max-upload-size = mkOption {
+        type = types.nullOr types.int;
+        default = null;
+      };
+
+      multi-user = mkOption {
+        type = types.bool;
+        default = false;
+      };
+
+      cuda-device = mkOption {
+        type = types.nullOr types.str;
+        description = ''The CUDA device to use.  Query for this using lspci or lshw.  Leave as null to auto-detect and/or use Nix CUDA settings (verify this before merging!).'';
```

Comment:
Is this still an open todo?

Reply:
Just verified it's not needed and ComfyUI will detect the device.  Thanks!